### DISCUSSION TITLE
Gift certificates

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2309] [Ideas]
> Created by: Eli VZ ([eli@elivz.com](eli@elivz.com)) on 2015/05/15 21:09:44 +0000
Votes at time of UserVoice import: 74

------
Would like to be able to create a "Gift Certificate" product (or product type) that would have flexible pricing (so the customer can set the amount they want to purchase), and then would generate a random code that would be associated with the product in order receipts, etc. Then there should be some way to redeem those codes at checkout as a payment (or partial-payment) method.

**Comment by andris-sevcenko** [comment]
> Created by: Bryan Garrant ([info@garrant.com](info@garrant.com)) on 2015/12/28 20:58:30 +0000


------
We really need this feature

**Comment by andris-sevcenko** [comment]
> Created by: Neil Bradley ([neilbradley@icloud.com](neilbradley@icloud.com)) on 2015/12/29 08:42:05 +0000


------
I would like to see this option too, but I think there should also be a screen in admin to see redeemed certificates and be able to search through them to see how much of the certificate amount is remaining etc.

This would be useful for example where a client has both an online and physical store and they might like the certificate to be used in the physical store too. With the admin screen they could check how much balance a customer has remaining to use.

**Comment by andris-sevcenko** [comment]
> Created by: Bryan Garrant ([info@garrant.com](info@garrant.com)) on 2015/12/30 02:13:49 +0000


------
Great idea Neil

**Comment by andris-sevcenko** [comment]
> Created by: Bryan Garrant ([info@garrant.com](info@garrant.com)) on 2015/12/30 02:15:46 +0000


------
Having an option to create fixed or flexible amount certificates would be great.

**Comment by andris-sevcenko** [comment]
> Created by: Brendan Underwood ([brendan@ipixel.com.au](brendan@ipixel.com.au)) on 2016/01/31 00:43:05 +0000


------
I know this is similar to Coupon Codes in its makeup, so I would also ask that we can create these as well through Commerce, as well as Coupon Code &#39;batches&#39; that all have the same properties but might only be one time use. We have the need to create a batch of 100 (or whatever number) Coupons that we can give to a customer to then use to purchase an item.

**Comment by andris-sevcenko** [comment]
> Created by: Anonymous ([mario@serieseight.com](mario@serieseight.com)) on 2017/05/23 16:57:29 +0000


------
yes please we 100% nee this feature

**Comment by andris-sevcenko** [comment]
> Created by: Jason Lee ([jason@jasonleelab.com](jason@jasonleelab.com)) on 2017/06/19 16:14:15 +0000


------
I need this feature for an upcoming project.

**Comment by andris-sevcenko** [comment]
> Created by: Andrew Armitage ([andrew@adigital.co.uk](andrew@adigital.co.uk)) on 2017/08/01 11:37:29 +0000


------
Would love to see this added to commerce.

**Comment by andris-sevcenko** [comment]
> Created by: Anonymous ([merryellen@liftdivision.com](merryellen@liftdivision.com)) on 2018/03/14 15:41:08 +0000


------
Very related, but not exactly the same, would be a credit balance associated with the user and the ability to adjust and work with a user&#39;s credit balance without the user having to purchase a gift certificate product in order to achieve it.

**Comment by engram-design** [comment]
For anyone interested, there's a plugin for this - https://verbb.io/craft-plugins/gift-voucher/features
